js/ViewportResizer.js

```diff
@@ -0,0 +1,223 @@
+    this._startHeight = height
+  }
+
+  get viewportWidth () {
+    return this._viewportWidth
+  }
+
+  set viewportWidth (height) {
+    this._viewportWidth = height
+  }
+
+  get viewportHeight () {
+    return this._viewportHeight
+  }
+
+  set viewportHeight (height) {
+    this._viewportHeight = height
+  }
+
+  get devices () {
+    return this.settings.devices
+  }
+
+  constructor (ctx, options = {}) {
+    this.$ctx = $(ctx)
+    this.settings = Object.assign({}, {
+      selectors: {
+        controls: 'StyleguideViewport',
+        example: 'StyleguideExample',
+        iframe: 'StyleguideExample-frame'
+      }
+    }, options)
+  }
+
+  init () {
+    let self = this
+    // bind actions for viewport controls
+    // bind width input field
+    // init the viewport on load
+    let $viewportResizer = this.$ctx.nextElementSibling.querySelector(`.${this.selectors.iframe}`)
+    let $widthInput = this.$ctx.querySelector(`.${this.selectors.controls}-width`)
+    let $heightInput = this.$ctx.querySelector(`.${this.selectors.controls}-height`)
+
+    self.updateViewportContainer($viewportResizer)
+    // also update the input field with width
+    self.updateInputs($widthInput, $heightInput)
+
+    $widthInput.addEventListener('focusout', function () {
+      self.updateViewport({width: this.value})
+    })
+    // bind height input field
+    $heightInput.addEventListener('focusout', function () {
+      self.updateViewport({height: this.value})
+    })
+    // bind function to the reset button
+    this.$ctx.querySelector(`.${this.selectors.controls}-reset`).addEventListener('click', () => {
+      this.updateViewport({width: '', height: ''})
+    })
+    // bind function based on preset viewports
+    this.$ctx.querySelectorAll('[data-viewportsize]').forEach(function (element) {
+      element.addEventListener('click', function () {
+        self.updateViewport(JSON.parse(this.getAttribute('data-viewportsize')))
+      })
+    })
+    // bind to Styleguide:updateViewport eventlistener
+    this.$ctx.addEventListener('Styleguide:updateViewport', function (event) {
+      self.updateViewportContainer($viewportResizer)
+      // also update the input field with width
+      self.updateInputs($widthInput, $heightInput)
+    })
+    // listens for tab change event and removes or adds width of viewport
+    this.$ctx.nextElementSibling.querySelector(`.${this.selectors.iframe}`).addEventListener('Styleguide:tabChange', function (event) {
+      if (window.location.hash === '#example') {
+        this.style.width = `${self.viewportWidth}px`
+        this.style.height = `${self.viewportHeight}px`
+      } else {
+        this.parentNode.removeAttribute('style')
+      }
+    })
+
+    // bind viewport frame actions
+    this.$ctx.nextElementSibling._.contents($.create('div', {
+      className: `${this.selectors.example}-handle`
+    }))
+
+    this.$ctx.nextElementSibling.addEventListener('mousedown', function (event) {
```

Comment:
Depending on which resize "handle" the user grabs, I think we should constrain the resize to that axis. For example: 
- Grabbing the vertical handle should resize constrained to the `x` axis.
- Grabbing the horizontal handle should resize constrained to the `y` axis.
- Grabbing the corner handle should (continue to) resize on both axes.

Also, the cursor should visually change according to the constraint instead of always being "omni-directional".